lib/shopping_cart.rb

```diff
@@ -0,0 +1,64 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products,
+              :sorted,
+              :breakdown
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @sorted = []
+    @breakdown = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    detail = {
+      name: "#{@name}",
+      capacity: @capacity}
+  end
+
+  def total_number_of_products
+    products.sum do |product|
+      product.quantity
+    end
+  end
+
+  def products_by_category(category)
+    products.select do |product|
+      product.category == category
+    end
+  end
+
+  def percentage_occupied
+    ((total_number_of_products.to_f / capacity.to_f) * 100).round(2)
+  end
+
+  def sorted_products_by_quantity
+    @sorted << products.sort_by do |product|
+      product.quantity
+    end
```

Comment:
Thsi is super close! if you reverse this return value, then you'll have what the interaction pattern is looking for.